lambdas/handlers/edge_presign_handler.py

```diff
@@ -13,44 +13,81 @@
 logger = logging.getLogger()
 logger.setLevel(logging.INFO)
 
+REQUIRED_QUERY_PARAMS = [
+    "X-Amz-Algorithm",
+    "X-Amz-Credential",
+    "X-Amz-Date",
+    "X-Amz-Expires",
+    "X-Amz-SignedHeaders",
+    "X-Amz-Signature",
+    "X-Amz-Security-Token",
+]
+
+REQUIRED_HEADERS = ["host", "cloudfront-viewer-country", "x-forwarded-for"]
+
 
 @set_request_context_for_logging
 @override_error_check
 @handle_edge_exceptions
```

Comment:
Looks like we have a decorator here but are not using it to it's full extent. From what I can see, accessing the `uri`, `query_string`, `headers`, `origin` and `domain_name` from the request can be extracted out to this decorator. This way the bulk of `try/excepts` are taken out of the handlers to make it clearer and easier to focus on the purpose of the handlers

Reply:
Good shout